Sources/SwiftParser/Patterns.swift

```diff
@@ -294,9 +325,14 @@ extension Parser.Lookahead {
       return true
     case (.lhs(.leftParen), _)?:
       return self.canParsePatternTuple()
-    case (.rhs, let handle)?:
-      self.eat(handle)
-      return self.canParsePattern()
+    case (.rhs(let introducer), let handle)?:
+      if shouldParsePatternBinding(introducer: introducer) {
+        self.eat(handle)
+        return self.canParsePattern()
+      } else {
+        self.eat(handle)
+        return true
```

Comment:
I was confused about this for a little bit until I read your comment in `parsePattern`. I think adding a comment like the following would make it clearer.

```swift
// The introducer doesn’t start a pattern binding. We should parse it as if it was an identifier.
```